dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java

```diff
@@ -173,6 +173,35 @@ public List<String> canWrite(User user, TrackedEntityInstance trackedEntity) {
     return canWrite(user, trackedEntity, programService.getAllPrograms());
   }
 
+  /**
+   * Check if the user can create the TE. For a regular user, they must have data write permissions
+   * to tracked entity type metadata as well as Capture Access to the Org Unit.
+   *
+   * @return No errors if a user has write access to the TrackedEntityType and to the OrgUnit
+   */
+  @Override
+  @Transactional(readOnly = true)
+  public List<String> canCreate(User user, TrackedEntityInstance trackedEntity) {
+    if (user == null || user.isSuper() || trackedEntity == null) {
```

Comment:
If I remember correctly, user was null when running internal jobs, but after the last refactor by the platform team, that's no longer the case

Reply:
This PR is for 2.39, and I also see other methods in the class doing the same user == null check. 
Unsure if the refactor you talk about exists in 2.39.

Reply:
Ah for some reason, your second comment just got in. Yes Agreed :D

Reply:
That a `null` user means `canCreate` == true does not feel right. If we assume that the user must be authenticated when this is called we should not check for null. Better an NPE than unauthorized access

Reply:
It doesn't feel right, but there's a job in previous versions run by a null user, which is why we need to skip this validation. However, it might be worth checking if that job needs to create a TE.

Reply:
I agree it does not feel right. Which is why in later versions, there was some work done to fix the user to never being null. 
But in 2.39, I feel this may be safer. To follow the same approach as is done for canUpdate , canDelete, and canRead.